diff --git a/drivers/media/platform/qcom/camss/camss-csid-680.c b/drivers/media/platform/qcom/camss/camss-csid-680.c
index 345a67c8fb94..ad612a7cfa98 100644
--- a/drivers/media/platform/qcom/camss/camss-csid-680.c
+++ b/drivers/media/platform/qcom/camss/camss-csid-680.c
@@ -268,13 +268,10 @@ static void __csid_configure_rdi_stream(struct csid_device *csid, u8 enable, u8
 	val |= dt_id << RDI_CFG0_DT_ID;
 	writel(val, csid->base + CSID_RDI_CFG0(port));
 
+	/* No crop or drop configuration is programmed for the RDI path. */
 	val = RDI_CFG1_TIMESTAMP_STB_FRAME;
 	val |= RDI_CFG1_BYTE_CNTR_EN;
 	val |= RDI_CFG1_TIMESTAMP_EN;
-	val |= RDI_CFG1_DROP_H_EN;
-	val |= RDI_CFG1_DROP_V_EN;
-	val |= RDI_CFG1_CROP_H_EN;
-	val |= RDI_CFG1_CROP_V_EN;
 	val |= RDI_CFG1_PACKING_MIPI;
 
 	writel(val, csid->base + CSID_RDI_CFG1(port));
diff --git a/drivers/media/platform/qcom/camss/camss-vfe-680.c b/drivers/media/platform/qcom/camss/camss-vfe-680.c
index 99036e7c1e76..0d228bc3032b 100644
--- a/drivers/media/platform/qcom/camss/camss-vfe-680.c
+++ b/drivers/media/platform/qcom/camss/camss-vfe-680.c
@@ -38,6 +38,7 @@
 
 #define VFE_BUS_WRITE_CLIENT_CFG(vfe, c)	((vfe_is_lite(vfe) ? 0x400 : 0xe00) + (c) * 0x100)
 #define		VFE_BUS_WRITE_CLIENT_CFG_EN	BIT(0)
+#define		VFE_BUS_WRITE_CLIENT_CFG_MODE_MIPI_RAW	BIT(16)
 #define VFE_BUS_IMAGE_ADDR(vfe, c)		((vfe_is_lite(vfe) ? 0x404 : 0xe04) + (c) * 0x100)
 #define VFE_BUS_FRAME_INCR(vfe, c)		((vfe_is_lite(vfe) ? 0x408 : 0xe08) + (c) * 0x100)
 #define VFE_BUS_IMAGE_CFG0(vfe, c)		((vfe_is_lite(vfe) ? 0x40c : 0xe0c) + (c) * 0x100)
@@ -186,7 +187,8 @@ static void vfe_wm_start(struct vfe_device *vfe, u8 rdi, struct vfe_line *line)
 	vfe_disable_irq(vfe);
 
 	/* Enable WM */
-	writel(VFE_BUS_WRITE_CLIENT_CFG_EN,
+	writel(VFE_BUS_WRITE_CLIENT_CFG_EN |
+	       VFE_BUS_WRITE_CLIENT_CFG_MODE_MIPI_RAW,
 	       vfe->base + VFE_BUS_WRITE_CLIENT_CFG(vfe, wm));
 
 	dev_dbg(vfe->camss->dev, "RDI%d WM:%d width %d height %d stride %d\n",
